### DISCUSSION TITLE
ValueError: Failed to submit verification request: Invalid API Key

### DISCUSSION THREAD

**Discussion by Kvicii** [smartcontractkit/full-blockchain-solidity-course-py#1604] [Q&A] [answered]
I want to verify FundMe contract, and 
1. I get the API-KEY on etherscan.io
2. config .env ETHERSCAN_TOKEN
3. deploy function add publish_source=True

but now deploy in rinkeby, there have some error, like this:
ValueError: Failed to submit verification request: Invalid API Key

I trying some other ways, but don't work, can you answer me

**Comment by cromewar** [comment]
Hello @Kvicii 
This could be for several reasons.
1. You might have a typo on the key.
2. For key is not activated yet.
3. your SO is not recognizing the kye yet (a terminal restart should work).

**Comment by Kvicii** [comment, marked as answer]
@cromewar hi, what should I do? I think the second one is reason

**Reply by cromewar** [reply]
well depends, please let me know the following:
1. which OS are you using?
2. generate a new api_key adn try again.
3. be sure the name key is the exact same as Patricks has on the course video.

**Reply by Kvicii** [reply]
1.MacOS
2.I retry three times, but no worked.
3.I config .env use export ETHERSCAN_TOKEN=xxxxxxx, and deploy function use publish_source=True

**Reply by cromewar** [reply]
as you are using a mac Os, could you please use the export line directly on the terminal? 
before running the deploy, copy and paste the entire line yo you terminal ` export ETHERSCAN_TOKEN=xxxxxxx` and press enter.
Try running it again, does it works?

**Reply by Kvicii** [reply]
seems not work, I use Brownie v1.19.0

**Reply by cromewar** [reply]
just a last verification, did you add `dotenv: .env` into brownie-config?
If nothing works seems for me a problem with EtherScan.

**Reply by Kvicii** [reply]
sure, I added. And yesterday I create API-KEY, but now also invalid, I'm not sure

**Reply by cromewar** [reply]
really weird, have your code on github? if so please share it so we can review it.

**Reply by Kvicii** [reply]
No, I intend find bug, If I don't resolve, I will deploy this tomorrow

**Reply by Kvicii** [reply]
ETH gov issue, thank you, I solved.

**Reply by Python-David** [reply]
I had this problem for days and finally here's what worked for me:

I changed my global compiler version to the version of solidity I declared for my contract: 0.6.6 (from top left...code > preferences > solidity > settings > search solidity > click on solidity configurations > compile using remote version OR just right click on the "pragma solidity..." and select "Change global compiler version")

For some reason, this worked. I'm guessing it has something to do with this from the brownie documentation - WARNING: Make sure all your source files use the same compiler version, otherwise the verification will fail.

**Reply by Kvicii** [reply]
@Python-David I just email to ETH.io, and they find problem, then I re-create it, it's work
